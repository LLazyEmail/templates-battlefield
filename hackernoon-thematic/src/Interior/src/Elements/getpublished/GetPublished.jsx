import React from 'react';

import { styles } from './styles';

import { link1, link2 } from './config';

import { Link } from 'react-emails-components-miscellaneous'; 

const GetPublished = () => (
  <div dir="ltr" style={styles.div}>
    <span style={styles.span1}>
      <span style={styles.span2}>
        <a href={link1}
          style={styles.a}
          target="_blank">
            Got a tech story to share with our readers?
        </a> 
        Everything you&#39;ve ever wanted to know about how to get published on Hacker Noon - 
        <a href={link2} style={styles.a} target="_blank">get it here</a>.
      </span>
    </span>
  </div>
);

export default GetPublished;