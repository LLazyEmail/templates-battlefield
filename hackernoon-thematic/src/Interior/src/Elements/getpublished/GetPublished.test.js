import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import GetPublished from './index';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('GetPublished/index component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GetPublished />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = create(<GetPublished />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Link has correct path', () => {
    const wrapper = shallow(<GetPublished />);
    const elem = wrapper.find('a');
    expect(elem.at(0).props()).toHaveProperty('href', 'https://hackernoon.com/signup');
    expect(elem.at(0).props()).toHaveProperty('target', '_blank');
  });

  it('Link text check', () => {
    const wrapper = shallow(<GetPublished />);
    const elem = wrapper.find('a');
    expect(elem.at(0).render().text()).toEqual("Got a tech story to share with our readers?");
  });

  it('Link text check(1)', () => {
    const wrapper = shallow(<GetPublished />);
    const elem = wrapper.find('a');
    expect(elem.at(1).props()).toHaveProperty('href', 'https://publish.hackernoon.com/');
    expect(elem.at(1).props()).toHaveProperty('target', '_blank');
  });

  it('Link has correct path(1)', () => {
    const wrapper = shallow(<GetPublished />);
    const elem = wrapper.find('a');
    expect(elem.at(1).render().text()).toEqual("get it here");
  });
});