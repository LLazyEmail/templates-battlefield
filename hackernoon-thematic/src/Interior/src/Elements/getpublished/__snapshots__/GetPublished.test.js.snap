// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetPublished/index component renders correctly 1`] = `
<div
  dir="ltr"
  style={
    Object {
      "textAlign": "justify",
    }
  }
>
  <span
    style={
      Object {
        "fontSize": "16px",
      }
    }
  >
    <span
      style={
        Object {
          "fontFamily": "trebuchet ms,lucida grande,lucida sans unicode,lucida sans,tahoma,sans-serif",
        }
      }
    >
      <a
        href="https://hackernoon.com/signup"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontWeight": "bold",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "textDecoration": "underline",
          }
        }
        target="_blank"
      >
        Got a tech story to share with our readers?
      </a>
       Everything you've ever wanted to know about how to get published on Hacker Noon - 
      <a
        href="https://publish.hackernoon.com/"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontWeight": "bold",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "textDecoration": "underline",
          }
        }
        target="_blank"
      >
        get it here
      </a>
      .
    </span>
  </span>
</div>
`;
